uaclient/jobs/gcp_auto_attach.py

```diff
@@ -27,15 +37,21 @@ def gcp_auto_attach(cfg: config.UAConfig) -> None:
     if cfg.is_attached:
         return
 
-    try:
-        # This function already uses the assert lock decorator,
-        # which means that we don't need to make create another
-        # lock only for the job
-        action_auto_attach(args=None, cfg=cfg)
-    except exceptions.NonAutoAttachImageError:
-        # If we get a NonAutoAttachImageError we know
-        # that the machine is not ready yet to perform an
-        # auto-attach operation (i.e. the license may not
-        # have been appended yet). If that happens, we will not
-        # error out.
-        return
+    # Only try to auto_attach if the license is found in the metadata
+    token = UAAutoAttachGCPInstance().identity_doc["identityToken"]
+    licenses = get_licenses_from_identity(token)
+    series = get_platform_info()["series"]
+
+    if GCP_LICENSES[series] in licenses:
```

Comment:
I think we need to be a bit more defensive here and explicit  as this will break on all non-LTS with a KeyError.
```suggestion
    if series in GCP_LICENSE and GCP_LICENSES[series] in licenses:
```

Reply:
We can check for it before, so we don't even need to go for metadata